Add idempotence as a first-class concept in best practices (#9473)
fishtown-analytics/docs.getdbt.com: website/docs/terms/hover-terms.md

idempotent:
  displayText: idempotent

>   hoverSnippet: An idempotent operation produces the same result no matter how many times you run it. In dbt, this means your models should be safe to re-run and always produce consistent results -- whether they've run once or a hundred times.

Reviewer: Seems a little long for a hover snippet, wonder if there's an opportunity to shorten it
```suggestion
  hoverSnippet: An idempotent operation produces the same results no matter how many times you run it. In dbt, this means your model runs will always produce consistent results.
```
